File: app/src/main/java/com/mastip/personaltodolist/support/adapter/viewholder/JTaskViewHolder.java
package com.mastip.personaltodolist.support.adapter.viewholder;

import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

import com.mastip.personaltodolist.R;
import com.mastip.personaltodolist.model.TaskModel;
import com.mastip.personaltodolist.utilities.Constant;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by HateLogcatError on 8/29/2017.
 */

public class JTaskViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.adapter_card_cardview)
    CardView card;
    @BindView(R.id.adapter_task_textview)
    TextView textTask;
    @BindView(R.id.adapter_time_textview)
    TextView textTime;
    @BindView(R.id.adapter_finish_textview)
    TextView textFinish;

    SimpleDateFormat simpleDateFormat = new SimpleDateFormat("dd MMMM yyyy", Locale.US);

    public JTaskViewHolder(View itemView) {
        super(itemView);

        ButterKnife.bind(this, itemView);
    }

    public void setUpModelToUI(Context context, TaskModel taskData) {
        textTask.setText(taskData.getTask());
        textTime.setText(context.getString(R.string.added) + simpleDateFormat.format(new Date(taskData.getTime())));
        textFinish.setText(context.getString(R.string.submission) + simpleDateFormat.format(new Date(taskData.getSubmission())));

        if(taskData.getPriority() == Constant.TAG_LOW_PRIORITY) {
            card.setCardBackgroundColor(ContextCompat.getColor(context, R.color.transparent_red));
        }
        else if(taskData.getPriority() == Constant.TAG_MEDIUM_PRIORITY) {
            card.setCardBackgroundColor(ContextCompat.getColor(context, R.color.transparent_yellow));
        }
        else if(taskData.getPriority() == Constant.TAG_HIGH_PRIORITY) {
            card.setCardBackgroundColor(ContextCompat.getColor(context, R.color.transparent_green));
        }
    }


}

File: app/src/main/java/com/mastip/personaltodolist/support/adapter/viewholder/JTaskViewHolder.kt
package com.mastip.personaltodolist.support.adapter.viewholder

import android.content.Context
import androidx.core.content.ContextCompat
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import android.view.View
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.mastip.personaltodolist.R
import com.mastip.personaltodolist.model.TaskModel
import com.mastip.personaltodolist.utilities.Constant
import java.text.SimpleDateFormat
import java.util.*

class JTaskViewHolder(itemView: View) : androidx.recyclerview.widget.RecyclerView.ViewHolder(itemView) {
    
    @BindView(R.id.adapter_card_cardview)
    lateinit var card: androidx.cardview.widget.CardView
    
    @BindView(R.id.adapter_task_textview)
    lateinit var textTask: TextView
    
    @BindView(R.id.adapter_time_textview)
    lateinit var textTime: TextView
    
    @BindView(R.id.adapter_finish_textview)
    lateinit var textFinish: TextView
    
    var simpleDateFormat = SimpleDateFormat("dd MMMM yyyy", Locale.US)
    
    fun setUpModelToUI(context: Context, taskData: TaskModel) {
        textTask.text = taskData.task
        textTime.text = context.getString(R.string.added) + simpleDateFormat.format(Date(taskData.time))
        textFinish.text = context.getString(R.string.submission) + simpleDateFormat.format(Date(taskData.submission))
        when (taskData.priority) {
            Constant.TAG_LOW_PRIORITY -> {
                card.setCardBackgroundColor(ContextCompat.getColor(context, R.color.transparent_red))
            }
            Constant.TAG_MEDIUM_PRIORITY -> {
                card.setCardBackgroundColor(ContextCompat.getColor(context, R.color.transparent_yellow))
            }
            Constant.TAG_HIGH_PRIORITY -> {
                card.setCardBackgroundColor(ContextCompat.getColor(context, R.color.transparent_green))
            }
        }
    }

    init {
        ButterKnife.bind(this, itemView)
    }
}